sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InMemoryFileIndex.scala

```diff
@@ -176,9 +180,27 @@ object InMemoryFileIndex extends Logging {
   private[sql] def bulkListLeafFiles(
       paths: Seq[Path],
       hadoopConf: Configuration,
-      filter: PathFilter,
+      filters: ParArray[PathFilter],
       sparkSession: SparkSession,
       areRootPaths: Boolean): Seq[(Path, Seq[FileStatus])] = {
+    bulkListLeafFiles(paths, hadoopConf, filters, sparkSession, areRootPaths, Map[String, String]())
```

Comment:
Classes leveraging InMemoryFileIndex continue using the same interface for bulkListLeafFiles, no risk for backwards compatibility issues.